src/test/java/com/appium/gesture/GestureTest.java

```diff
@@ -166,22 +166,35 @@ private void verticalSwipe(String locator) throws InterruptedException {
     @Test
     public void multiTouchTest() throws InterruptedException {
         login();
-        wait.until(ExpectedConditions.
-                elementToBeClickable(MobileBy.AccessibilityId("slider1"))).click();
+        Thread.sleep(2000);
+        wait.until(ExpectedConditions.presenceOfElementLocated(MobileBy.AccessibilityId("slider1"))).click();
         Thread.sleep(3000);
         MobileElement slider = driver.findElementByAccessibilityId("slider");
         MobileElement slider1 = driver.findElementByAccessibilityId("slider1");
 
-        Dimension sizeSlider = slider.getSize();
-        Dimension sizeSlider1 = slider1.getSize();
-        TouchAction touchAction1 =
-                new TouchAction(driver).press(ElementOption.element(slider, 0, sizeSlider.height / 2))
-                        .waitAction(WaitOptions.waitOptions(Duration.ofSeconds(1)))
-                        .moveTo(ElementOption.element(slider, sizeSlider.width / 2, sizeSlider.height / 2));
-        TouchAction touchAction2 =
-                new TouchAction(driver).press(ElementOption.element(slider1, 0, sizeSlider1.height / 2)).waitAction(WaitOptions.waitOptions(Duration.ofSeconds(1)))
-                        .moveTo(ElementOption.element(slider1, sizeSlider1.width / 2, sizeSlider1.height / 2));
-        new MultiTouchAction(driver).add(touchAction1).add(touchAction2).perform();
-        Thread.sleep(2000);
+        Point source1 = slider.getLocation();
+        Point source2 = slider1.getLocation();
+
+        PointerInput pointerInput1 = new PointerInput(PointerInput.Kind.TOUCH, "finger1");
```

Comment:
Are you sure `PointerInput` accepts it as `finger1`?

Reply:
Yes it works

Reply:
@SrinivasanTarget Selenium handles it 

    public PointerInput(PointerInput.Kind kind, String name) {
        this.kind = (PointerInput.Kind)Objects.requireNonNull(kind, "Must set kind of pointer device");
        this.name = (String)Optional.ofNullable(name).orElse(UUID.randomUUID().toString());
    }